acq_stat_reports/utils.py

```diff
@@ -0,0 +1,538 @@
+import functools
+import itertools
+from pathlib import Path
+from typing import Any
+
+# from typing import Callable
+import matplotlib.pyplot as plt
+import numpy as np
+
+# import numpy.typing as npt
+from astropy.table import Table
+from chandra_aca.star_probs import binomial_confidence_interval
+
+from acq_stat_reports.config import conf
+
+
+class BinnedData:
+    def __init__(self, data, bins, extra_cols=None, selection=None):
+        self.bins = bins
+        self.selection = selection if selection is not None else (lambda x: x)
+        self.data = self.selection(data)
```

Comment:
It's not intuitive that the `data` attribute gets auto-filtered.
```
bd = BinnedData(data, bins, selection=...)
# unexpected that bd.data is different from data.
```

Reply:
By default it was not auto-filtered. It was auto-filtered if you gave it the option, in which case it would be pretty much what you expect.

Anyway, I removed the selection in [85ea8a](https://github.com/sot/acq_stat_reports/pull/6/commits/85ea8a41578d21eb98a34e16ce9db513699b6885)